### PR TITLE
Adds auto-generated documentation from JSON Schemas

### PR DESCRIPTION
**Pull request by p-galligan**
Adds HTML documentation of the JSON schemas, as well as a script to generate that documentation from JSON files.

### CODE REVIEW

File: generate_docs.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import argparse
+import os
+
+from json_schema_for_humans.generate import generate_from_filename
+
+
+class DocumentGenerator:
+    '''Generates HTML documentation based on JSON schema files.'''
```
Comment (helrond) [inline generate_docs.py]: Double quotes in docstrings plz!

File: generate_docs.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+
+parser = argparse.ArgumentParser(
+    description="Creates HTML files based on JSON schemas.")
+parser.add_argument(
+    "json_directory",
+    help="Directory containing JSON Schema files.")
+parser.add_argument(
+    "html_directory",
+    help="Directory to write HTML files to.")
```
Comment (helrond) [inline generate_docs.py]: Wondering if we can provide default values for these args, so the script without having to worry about input/output dirs?

File: generate_docs.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    '''Generates HTML documentation based on JSON schema files.'''
+
+    def run(self, json_directory, html_directory):
+        '''Iterates over a directory and generates an HTML schema file from any file ending in .json
+
+        args:
+            directory(str): a string representation of the full filepath of the directory containing schema files
+        '''
+        for file in os.listdir(json_directory):
+            if file.endswith(".json"):
```
Comment (helrond) [inline generate_docs.py]: You could do this in a single line using list comprehension.

**Review by helrond — CHANGES_REQUESTED** [review]
OK I've added some specific comments in here, but I think we also need to add a `Documentation` section to the README which describes how to generate docs (and the necessary dependencies).

File: generate_docs.py (hunk 3 above)
Comment (p-galligan) [reply generate_docs.py]: Should be done now!

**Review by helrond — APPROVED** [review]

**Comment by helrond** [comment]
Looking at the script a bit more, I think using a class-based approach is overly verbose and not really helpful since we're not passing any arguments, so I converted to a simple function instead, and added a module-level docstring.

**Pull request merged**